packages/next/src/loader.ts

```diff
@@ -60,15 +60,23 @@ export default async function workflowLoader(
     relativeFilename = normalizedFilepath.split('/').pop() || 'unknown.ts';
   }
 
+  const isJsx = isTsx || filename.endsWith('.jsx');
+
   // Transform with SWC
   const result = await transform(normalizedSource, {
     filename: relativeFilename,
     jsc: {
       parser: {
         syntax: isTypeScript ? 'typescript' : 'ecmascript',
         tsx: isTsx,
+        jsx: !isTypeScript && isJsx,
       },
       target: 'es2022',
+      transform: {
+        react: {
+          runtime: 'automatic',
```

Comment:
Ideally we don't actually transform JSX at all so that the upstream tool can consume how it needs to e.g. Next.js might process one way but then vite expects to process a different way. Let me see if I can get that to work